### ISSUE TITLE
clj-kondo throws exception on type-hinting and protocol arity

### ISSUE THREAD

**Issue by pithyless** [borkdude/clj-kondo#314]
**version**

clj-kondo v2019.06.29-alpha

**platform**

MacOS (via homebrew)

**problem**

clj-kondo crashes with error: `java.lang.IllegalStateException: Can't pop empty vector`

**repro**

This crash was discovered running clj-kondo on a large codebase. (Side note: there was no information what classpath was causing the issue; this meant I needed to manually bisect the classpath until I found a minimal case. Maybe this can be improved in the future?) 

I've managed to isolate it to the following repro (borrowing some code from `muuntaja` which was on the classpath):

```
echo '(ns muuntaja.protocols)

(defprotocol IntoInputStream
  (into-input-stream ^java.io.InputStream [this]))

(ns muuntaja.core
  (:require [muuntaja.protocols :as protocols]))

(defn slurp [x]
  (some-> x protocols/into-input-stream clojure.core/slurp))
' | clj-kondo --lint -
```

**expected behavior**

clj-kondo shows linter warnings :)

**actual behavior**

```
Unexpected error. Please report an issue.
java.lang.IllegalStateException: Can't pop empty vector
	at clojure.lang.PersistentVector.pop(PersistentVector.java:470)
	at clojure.lang.PersistentVector.pop(PersistentVector.java:22)
	at clojure.lang.RT.pop(RT.java:755)
	at clojure.core$pop.invokeStatic(core.clj:1474)
	at clj_kondo.impl.linters$show_arities.invokeStatic(linters.clj:141)
	at clj_kondo.impl.linters$arity_error.invokeStatic(linters.clj:150)
	at clj_kondo.impl.linters$lint_calls$iter__2742__2752$fn__2753$iter__2744__2754$fn__2755$iter__2746__2756$fn__2757.invoke(linters.clj:214)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clj_kondo.impl.linters$lint_calls$iter__2742__2752$fn__2753$iter__2744__2754$fn__2755.invoke(linters.clj:158)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clj_kondo.impl.linters$lint_calls$iter__2742__2752$fn__2753.invoke(linters.clj:158)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clojure.core$dorun.invokeStatic(core.clj:3133)
	at clojure.core$doall.invokeStatic(core.clj:3148)
	at clj_kondo.core$run_BANG_.invokeStatic(core.clj:94)
	at clj_kondo.main$main.invokeStatic(main.clj:75)
	at clj_kondo.main$main.doInvoke(main.clj:68)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clj_kondo.main$_main$fn__3557.invoke(main.clj:94)
	at clj_kondo.main$_main.invokeStatic(main.clj:94)
	at clj_kondo.main$_main.doInvoke(main.clj:92)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj_kondo.main.main(Unknown Source)
```

**possible culprits**

1. Linter works fine if we remove type hint:

```
echo '(ns muuntaja.protocols)

(defprotocol IntoInputStream
  (into-input-stream [this]))

(ns muuntaja.core
  (:require [muuntaja.protocols :as protocols]))

(defn slurp [x]
  (some-> x protocols/into-input-stream clojure.core/slurp))

' | clj-kondo --lint -

<stdin>:9:1: warning: slurp already refers to #'clojure.core/slurp
linting took 9ms, errors: 0, warnings: 1
```

2. Linter works fine if we keep type hint, but only process first namespace:

```
echo '(ns muuntaja.protocols)

(defprotocol IntoInputStream
  (into-input-stream ^java.io.InputStream [this]))
' | clj-kondo --lint -

linting took 8ms, errors: 0, warnings: 0
```

**Comment by pithyless** [comment]
One more datapoint: this works in the previously released version.

```
❯ /tmp/clj-kondo --version
clj-kondo v2019.06.23-alpha

❯ echo '(ns muuntaja.protocols)

(defprotocol IntoInputStream
  (into-input-stream ^java.io.InputStream [this]))

(ns muuntaja.core
  (:require [muuntaja.protocols :as protocols]))

(defn slurp [x]
  (some-> x protocols/into-input-stream clojure.core/slurp))

' | /tmp/clj-kondo --lint -

<stdin>:9:1: warning: slurp already refers to #'clojure.core/slurp
<stdin>:10:13: error: wrong number of args (1) passed to muuntaja.protocols/into-input-stream
linting took 8ms, errors: 1, warnings: 1
```

**Comment by borkdude** [comment]
@pithyless Thanks for the detailed report, I'll look into this and make sure it will be fixed in the next version.

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/9593ab11e0a09c7bb731f419784a32837e59dbe1

**Comment by borkdude** [comment]
@pithyless If you want to test, you can do this with:

```
clj -Sdeps '{:deps {clj-kondo {:git/url "https://github.com/borkdude/clj-kondo" :sha "9593ab11e0a09c7bb731f419784a32837e59dbe1"}}}' -A:clj-kondo --lint /tmp/code.clj
```

or via this new Mac binary:

https://4248-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.06.30-alpha-SNAPSHOT-macos-amd64.zip

**Comment by pithyless** [comment]
@borkdude confirming it works, thanks for the amazing response time! :)
